include/flatbuffers/flatbuffers.h

```diff
@@ -377,22 +377,34 @@ class Allocator {
   }
 };
 
-// DefaultAllocator uses new/delete to allocate memory regions
-class DefaultAllocator : public Allocator {
- public:
-  virtual uint8_t *allocate(size_t size) FLATBUFFERS_OVERRIDE {
+inline uint8_t *Allocate(Allocator *allocator, size_t size) {
+  if (allocator) {
+    return allocator->allocate(size);
+  } else {
     return new uint8_t[size];
   }
+}
 
-  virtual void deallocate(uint8_t *p, size_t) FLATBUFFERS_OVERRIDE {
+inline void Deallocate(Allocator *allocator, uint8_t *p, size_t size) {
+  if (allocator) {
+    return allocator->deallocate(p, size);
+  } else {
     delete[] p;
   }
+}
 
-  static DefaultAllocator &instance() {
-    static DefaultAllocator inst;
-    return inst;
+inline uint8_t *ReallocateDownward(Allocator *allocator, uint8_t *old_p,
+                                   size_t old_size, size_t new_size) {
+  if (allocator) {
+    return allocator->reallocate_downward(old_p, old_size, new_size);
+  } else {
+    assert(new_size > old_size);  // vector_downward only grows
+    uint8_t *new_p = new uint8_t[new_size];
+    memmove(new_p + (new_size - old_size), old_p, old_size);
```

Comment:
These 5 lines are copied from `Allocator::reallocate_downward`.. either it needs to refer to that function, or that function should be removed?